### ISSUE TITLE
LC0012 - False positive if procedure-call is used as argument to statements like Page.RunModal

### ISSUE THREAD

**Issue by rvanbekkum** [StefanMaron/BusinessCentral.LinterCop#78]
Hi @StefanMaron,

Just ran into the following false positive for LC0012. I only expect this rule to give a false positive if a constant integer is used.
It also throws a warning if a procedure-call is used as an argument.

Example:
```al
local procedure RunSetupPage()
begin
    Page.RunModal(GetSetupPage());
end;

local procedure GetSetupPage(): Integer
begin
    exit(Page::"General Ledger Setup");
end;
```

![image](https://user-images.githubusercontent.com/7383241/140524352-99f5f052-0f83-41cd-9737-ac1055798232.png)

**Comment by StefanMaron** [comment]
Ah I did not think about procedure calls. Thanks for reporting!

**Comment by dnpb** [comment]
It could also happen via variables, e.g. `SomePageId: Integer` is set conditionally in a `case` statement, then used to `Page.Run(SomePageId, PurchaseHeader)`.

**Comment by rvanbekkum** [comment]
I think the update for this is to be released, right? 😉

**Comment by StefanMaron** [comment]
Yes, it was closed by accident because of the link to the pull request
/open
